File: webui/src/main/scala/com/taskadapter/webui/config/ApplicationSettings.scala
package com.taskadapter.webui.config

import java.io.File

object ApplicationSettings {
  /**
    * Calculates default taskadapter root folder. Current implementation
    * returns <code>user.home/.taskadapter</code>.
    *
    * @return task adapter config folder.
    */
  def getDefaultRootFolder(): File = {
    val userHome = System.getProperty("user.home")
    new File(userHome, ".taskadapter")
  }
}

File: webui/src/main/java/com/taskadapter/webui/config/ApplicationSettings.java
package com.taskadapter.webui.config;

import java.io.File;

public class ApplicationSettings {

    /**
     * Calculates default taskadapter root folder. Current implementation
     * returns <code>user.home/.taskadapter</code>.
     *
     * @return task adapter config folder.
     */
    public static File getDefaultRootFolder() {
        var userHome = System.getProperty("user.home");
        return new File(userHome, ".taskadapter");
    }
}
